src-plugins/mscExportVideo/mscExportVideo.cpp

```diff
@@ -0,0 +1,307 @@
+// ExportVideoPrivate
+// /////////////////////////////////////////////////////////////////
+namespace msc
+{
+class ExportVideoPrivate
+{
+public:
+    QVector< QImage > imagesArray;
+    int width;
+    int height;
+    int maximumFrame;
+
+    // User parameters
+    QString filename;
+    int format;            // Cf. VideoFormat in header file
+    int frameRate;         // Frame per second
+    bool subsampling;      // Is the video to be encoded using 4:2:0 subsampling?
+    int quality;           // 0 = low, 1 = medium, 2 = high
+    bool exportOnlyAsJPEG; // Can be used for home encoding
+};
+
+// /////////////////////////////////////////////////////////////////
+// ExportVideo
+// /////////////////////////////////////////////////////////////////
+
+ExportVideo::ExportVideo() : medAbstractProcess(), d(new ExportVideoPrivate)
+{
+    d->imagesArray.clear();
+    d->width  = 0;
+    d->height = 0;
+    d->maximumFrame = 0;
+
+    // User parameters
+    d->format = OGGVORBIS;
+    d->filename = "";
+    d->frameRate = 1;
+    d->subsampling = false;
+    d->quality = 2;
+    d->exportOnlyAsJPEG = false;
+}
+
+ExportVideo::~ExportVideo()
+{
+    delete d;
+}
+
+bool ExportVideo::registered()
+{
+    return dtkAbstractProcessFactory::instance()->registerProcessType("msc::ExportVideo", createExportVideo);
+}
+
+QString ExportVideo::description() const
+{
+    return "ExportVideo";
+}
+
+QString ExportVideo::identifier() const
+{
+    return description();
+}
+
+void ExportVideo::setParameter(int* data, int frame)
+{
+    d->width  = data[0];
+    d->height = data[1];
+
+    QImage currentImage = QPixmap(d->width, d->height).toImage();
+
+    for (int i=0; i<d->width; ++i)
+    {
+        for (int j=0; j<d->height; ++j)
+        {
+            // R, G, B to QColor (A default at 255)
+            int index1D = 2 + (i * d->height + j)*3;
+            QColor color = QColor(data[index1D    ],
+                                  data[index1D + 1],
+                                  data[index1D + 2]);
+
+            currentImage.setPixel(i, j, color.rgb());
+        }
+    }
+
+    int arraySize = d->imagesArray.size();
+
+    while (arraySize <= frame)
+    {
+       d->imagesArray.resize(1 + 2 * arraySize); // it's common to double here to avoid N resizes when giving N frames in order
+       arraySize = d->imagesArray.size();
+    }
+
+    d->imagesArray[frame] = currentImage;
+    d->maximumFrame = std::max(d->maximumFrame, frame);
+}
+
+medAbstractData* ExportVideo::output()
+{
+    return nullptr;
+}
+
+int ExportVideo::update()
+{
+    if (!d->imagesArray.empty())
+    {
+        // Handle error of missing frame: ignore the frame by removing it
+        d->imagesArray.resize(d->maximumFrame);
+        int cpt = 0;
+        foreach (QImage image, d->imagesArray)
+        {
+            if (image.isNull())
+            {
+                d->imagesArray.remove(cpt);
+            }
+            cpt++;
```

Comment:
This line needs to be in an ```else``` statement because when removing an element from the array the indices of the remaining elements are all decremented by 1.